ext/scheduler/airflow/__lib.py

```diff
@@ -470,22 +469,25 @@ def optimus_sla_miss_notify(dag, task_list, blocking_task_list, slas, blocking_t
         }
         # post event
         resp = optimus_client.notify_event(params["project_name"], params["namespace"], params["job_name"], event)
-        print("posted event ", params, event, resp)
+        log.info(f'posted event {params}, {event}, {resp}')
         return
     except Exception as e:
         print(e)
 
 def shouldSendSensorStartEvent(ctx):
     try:
+        from airflow import settings
         ti = ctx.get('task_instance')
         key = "sensorEvt/{}/{}/{}".format(ti.task_id , ctx.get('next_execution_date').strftime(TIMESTAMP_FORMAT) , ti.try_number)
-
-        result = ti.xcom_pull(key=key)
```

Comment:
need to understand why ti.xcom_pull did not work

Reply:
it it possible that before every ti schedule, xcom are deleted as per https://github.com/apache/airflow/blob/2.1.4/airflow/models/taskinstance.py#L1243

Reply:
![image](https://github.com/goto/optimus/assets/13954513/a294d673-a396-4fc3-b419-81532765f053)

you are right, it seems airflow is exhibiting strange inconsistent behaviour. This method has worked in past but dosent seem to be working now.
hence we are choosing to move to a diffrent approach